src/main/java/mate/academy/webapp/repository/book/BookRepository.java

```diff
@@ -0,0 +1,23 @@
+package mate.academy.webapp.repository.book;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.webapp.model.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+
+public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Pageable pageable);
+
+    @EntityGraph(attributePaths = "categories")
+    Optional<Book> findById(Long id);
+
+    @Query(value = "SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoryId")
```

Comment:
Is it possible to use `@EntityGraph` here, and generate query by method name? I think it should work too.